### PR TITLE
TSPS-370 Make pipelineVersion acceptable and not required 

### PR DESCRIPTION
**Pull request by jsotobroad**
### Description 
Made changes regarding pipeline version
* submit endpoint no longer requires a pipeline version and the service will default to the latest version
* pipeline details now takes an optional pipeline version input, also defaults to latest
* pipeline details and pipelines list now return pipeline version 

Note this will require https://github.com/DataBiosphere/terra-scientific-pipelines-service/pull/171 to be merged before it will be working

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-370

Pipelines

Screenshot 2024-12-02 at 2 30 29 PM
Screenshot 2024-12-02 at 2 30 48 PM
Screenshot 2024-12-02 at 2 30 57 PM
Screenshot 2024-12-02 at 2 31 05 PM

Submit

Screenshot 2024-12-02 at 2 31 41 PM
Screenshot 2024-12-02 at 2 31 55 PM
Screenshot 2024-12-02 at 2 32 09 PM
Screenshot 2024-12-02 at 2 32 31 PM

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,8 @@ terralab = "terralab.cli:cli"
 
 [tool.poetry.dependencies]
 python = "^3.12"
-terra-scientific-pipelines-service-api-client = "0.1.11"
+#terra-scientific-pipelines-service-api-client = "0.1.11"
```
Comment (jsotobroad) [inline pyproject.toml:17]: revert

File: terralab/.terralab-cli-config [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Teaspoons service API URL
-TEASPOONS_API_URL=https://teaspoons.dsde-dev.broadinstitute.org
+#TEASPOONS_API_URL=https://teaspoons.dsde-dev.broadinstitute.org
+TEASPOONS_API_URL=http://localhost:8080
```
Comment (jsotobroad) [inline terralab/.terralab-cli-config:3]: revert

File: terralab/commands/pipeline_runs_commands.py [hunk 3]
```diff
@@ -25,9 +25,7 @@
 
 @click.command(short_help="Submit a job")
 @click.argument("pipeline_name", type=str)
-@click.option(
-    "--version", type=int, default=0, help="pipeline version; default: 0"
-)  # once TSPS-370 is done, remove default
+@click.option("--version", type=int, help="pipeline version")
```
Comment (mmorgantaylor) [inline terralab/commands/pipeline_runs_commands.py:28]: can we include "defaults to latest" in the help text?

File: terralab/commands/pipelines_commands.py [hunk 4]
```diff
@@ -26,22 +26,27 @@ def list():
...
+@click.option("--version", type=int, help="pipeline version")
 @handle_api_exceptions
-def details(pipeline_name: str):
+def details(pipeline_name: str, version: int):
     """Get information about the PIPELINE_NAME pipeline"""
-    pipeline_info = pipelines_logic.get_pipeline_info(pipeline_name)
+    pipeline_info = pipelines_logic.get_pipeline_info(pipeline_name, version)
 
     # format the information nicely
     col_width = 16
```
Comment (mmorgantaylor) [inline terralab/commands/pipelines_commands.py:42]: need to up this value to 18 it looks like to get this new output aligned properly

**Review by mmorgantaylor — APPROVED** [review]
a couple minor things but looks good! woooohoo!

**Pull request merged**